src/Controller/GalleryController.php

```diff
@@ -31,7 +33,52 @@ public function showImagesByCategorie(string $categorie, ImageRepository $imageR
     }
 
     /**
-     * @Route("/gallerie-passages-secrets", name="gallerie-passages-secrets")
+     * @Route("/gallerie-design-metallique/{categorie}/{id}",
+     *          name="gallery_metallicDesign_category_showImg", methods={"GET"})
+     */
+    public function showImg(Image $image): Response
+    {
+        return $this->render('gallery/showImg.html.twig', [
+            'image' => $image,
+        ]);
+    }
+
+    /**
+     * @Route("/gallerie-design-metallique/{categorie}/{id}/edit",
+     *      name="gallery_metallicDesign_category_showImg_edit", methods={"GET","POST"})
+     */
+    public function edit(Request $request, Image $image): Response
+    {
+        $form = $this->createForm(EditFormType::class, $image);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->getDoctrine()->getManager()->flush();
```

Comment:
It is a symfony best practice to use dependency injection for services (like the EntityManager)